metric-schema-lang/src/main/java/com/palantir/metric/schema/lang/TagDefinition.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.metric.schema.lang;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Set;
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+public interface TagDefinition {
+    String tagName();
+
+    Set<String> values();
```

Comment:
It would be helpful if we had the option to describe each tag, perhaps even each predefined tag value. What do you think?

Reply:
I could see it being a nice follow up :)

Reply:
Of course, however it does have some bearing on the `metric-schema-api.yml` change to allow us to support more expressive syntax without breaking consumers.

Reply:
Sorry, I may have been overly terse. I believe we can provide that functionality in a forward compat way by doing the same thing that this PR does with tags. I.e Supporting the following:

```
namespaces:
  monitors:
    docs: General web server metrics.
    metrics:
      processing:
        docs: Measures number of installations that were processed
        type: meter
        tags:
          - result: [success, failure]
          - type
          - documentedTag:
            values:
              - value: a
                docs: Some docs about tag
              - b
            docs: This tag has documentation!
```

Reply:
Right, the `-lang` component should be straightforward, but we need to ensure the IR has a way to represent new data without deprecating a tags field and adding another each time we want to add a docs field.

Reply:
okiedokes, I misunderstood. Should be fixed up